src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -1248,11 +1248,17 @@ protected override int ExecuteCmd(DocumentsOperationContext context)
                     var operationsCount = 0;
 
                     var database = _replicationInfo.DocumentDatabase;
+                    var lastTransactionMarker = 0;
 
                     context.LastDatabaseChangeVector = context.LastDatabaseChangeVector ?? DocumentsStorage.GetDatabaseChangeVector(context);
                     foreach (var item in _replicationInfo.ReplicatedItems)
                     {
-                        context.TransactionMarkerOffset = item.TransactionMarker;
+                        if (lastTransactionMarker != item.TransactionMarker)
+                        {
+                            context.TransactionMarkerOffset++;
```

Comment:
Why do we need that?

Reply:
`TransactionMarkerOffset` is to distinguish between transaction when they get merged by the tx merger.
Incoming replication can have a batch that contains several transactions, and we need to distinguish between them as well.

The fix here is in case this command is part of merged tx, we don't want to override the existing `TransactionMarkerOffset`.